docs/infrastructure/native-image.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: developer-doc
+title: Native Image
+category: infrastructure
+tags: [infrastructure, build, native, native-image]
+order: 3
+---
+
+# Native Image
+
+[`NativeImage`](../../project/NativeImage.scala) defines a task that is used for
+compiling a project into a native binary using Graal's Native Image. It compiles
+the project and runs the Native Image tool which builds the image. Currently,
+Native Image is used for building the Launcher.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Requirements](#requirements)
+  - [Native Image Component](#native-image-component)
+  - [Additional Linux Dependencies](#additional-linux-dependencies)
+- [Static Builds](#static-builds)
+- [No Cross-Compilation](#no-cross-compilation)
+
+<!-- /MarkdownTOC -->
+
+## Requirements
+
+### Native Image Component
+
+The Native Image component has to be installed within the used GraalVM
+distribution. It can be installed by running `gu install native-image`.
```

Comment:
It is already mentioned [there](https://github.com/enso-org/enso/blob/wip/rw/launcher-version-management/docs/CONTRIBUTING.md#building-the-launcher-native-binary).

And the [check](https://github.com/enso-org/enso/blob/wip/rw/launcher-version-management/project/NativeImage.scala#L42-L51) is also present.

They were added much earlier in a previous PR preparing the Native Image build. This documentation has just been moved to a separate file in this PR because it has grew significantly with the additional information on `libc` and reflection configuration.